src/main/java/org/jenkinsci/plugins/workflow/log/FileLogStorage.java

```diff
@@ -126,17 +126,28 @@ private void checkId(String id) throws IOException {
         assert Thread.holdsLock(this);
         if (!Objects.equals(id, lastId)) {
             bos.flush();
-            long pos = os.getChannel().position();
-            if (id == null) {
-                indexOs.write(pos + "\n");
-            } else {
-                indexOs.write(pos + " " + id + "\n");
+            // FileChannel methods such as position() close the channel (and the stream it was derived from) if the
+            // current thread is interrupted. We must not allow interruption on some arbitrary thread attempting to
+            // write to the log to close the stream irrevocably, so we clear the interrupt status before using the
+            // channel and then restore the interrupt status afterwards.
+            boolean interrupted = Thread.interrupted();
+            try {
+                long pos = os.getChannel().position();
```

Comment:
Not perfect, because we do not mask interruptions that happen during the call to `position()`, but maybe this is good enough for most practical cases? We could check for closure of the channel and recreate `os` in that case and then recheck the position, but I am not sure if there is a way to make this code robust against repeated interruption.

Reply:
> track the position manually

Using https://javadoc.io/static/commons-io/commons-io/2.13.0/org/apache/commons/io/output/CountingOutputStream.html I suppose? You would however need to check the file length when opening `os` in append mode to get an initial offset.

Reply:
Yeah, it is a bit unfortunate because I think `FileOutputStream` has everything needed to read the position from the file descriptor in the same way that `FileChannel.position()` does, but there is no public API to support it.

Reply:
Also I don't think we need an `AtomicLong`, just a `long`, since all of this code is in `synchronized` blocks.

Reply:
:thinking: not quite, since those methods do not flush `bos`, so there would be a small race condition. Really want something like

```java
cos = new CountingOutputStream(os);
bos = new GCFlushedOutputStream(new DelayBufferedOutputStream(cos));
```

to track when bytes were actually flushed.

Reply:
See [e611d9c](https://github.com/jenkinsci/workflow-api-plugin/pull/296/commits/e611d9c3ffa8627cab1a50a774b60ec3266c56f8).